Assets/FullSerializer/Source/fsConfig.cs

```diff
@@ -53,7 +53,12 @@ public class fsConfig {
         public Func<string, MemberInfo, string> GetJsonNameFromMemberName = (name, info) => name;
 
         /// <summary>
-        /// Should the default serialization behaviour include non-auto properties?
+        /// Should the default serialization behaviour include any properties?
+        /// </summary>
+        public bool SerializeProperties = true;
```

Comment:
How about renaming `SerializeProperties` to `EnablePropertySerialization`, with a comment along the lines of

```c#
/// <summary>
/// If false, then *all* property serialization support will be disabled - even properties
/// explicitly annotated with fsProperty or any other opt-in annotation. 
///
/// Setting this to false means that SerializeNonAutoProperties and
/// SerializeNonPublicSetProperties will be completely ignored.
/// </summary>
public bool EnablePropertySerialization = true;
```

I'm also fine with the enum you proposed above, though I agree it is a bit messy. Maybe rename `PropertySerializationConfig` to `DefaultPropertySerialization`?